modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/ast/validation/ValidationUtils.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.artifact.activation.internal.ast.validation;
+
+import static org.mule.runtime.api.i18n.I18nMessageFactory.createStaticMessage;
+import static org.mule.runtime.ast.api.validation.Validation.Level.ERROR;
+import static org.mule.runtime.ast.api.validation.Validation.Level.WARN;
+
+import static java.lang.System.lineSeparator;
+import static java.util.stream.Collectors.joining;
+
+import org.mule.runtime.ast.api.validation.ArtifactAstValidator;
+import org.mule.runtime.ast.api.validation.ValidationResult;
+import org.mule.runtime.ast.api.validation.ValidationResultItem;
+import org.mule.runtime.core.api.config.ConfigurationException;
+
+import java.util.Collection;
+
+import org.slf4j.Logger;
+
+/**
+ * Utilities for handling {@link ValidationResult}s in a uniform way.
+ *
+ * @since 4.5.0
+ */
+public class ValidationUtils {
+
+  private ValidationUtils() {
+    // Empty private constructor to avoid instantiation.
+  }
+
+  /**
+   * Handles a {@link ValidationResult} uniformly. Warnings are logged, errors raise exceptions.
+   *
+   * @param validationResult The {@link ValidationResult} obtained using an {@link ArtifactAstValidator}.
+   * @param logger           The {@link Logger} to use for logging.
+   * @throws ConfigurationException If there was any result item with {@link ERROR} level.
+   */
+  public static void handleValidationResult(ValidationResult validationResult, Logger logger)
```

Comment:
Here I have a couple of options.

1. What this method is doing is quite generic, so I could split it in two: `logWarnings` and `throwIfContainsErrors`.
Since it is only used in two places, it sounds manageable, but my concern would be that most often than not, we would like to do both things together, so we would be repeating the pattern of calling `logWarnings` followed by `throwIfContainsErrors`. 
2. We could keep it as one method but rename it to something more explicit like `logWarningsAndThrowIfContainsErrors`. It is borderline verbose, but could work.
3. I'm not so convinced that this method "isn't handling anything". It takes the results of the validation and performs an action appropriate for each of the items (or in aggregation). If we go too over-specific on the method name we would be more limited in how we can change what we do with the items in the future without breaking the semantics.

Reply:
Renamed to `logWarningsAndThrowIfContainsErrors`.